README.md

```diff
@@ -11,572 +11,528 @@
-import qualified Streamly.FileSystem.Handle as FH
-import qualified System.IO as FH
-
-import Data.Char (ord)
-import System.Environment (getArgs)
-import System.IO (openFile, IOMode(..), stdout)
-
-withArg f = do
-    (name : _) <- getArgs
-    src <- openFile name ReadMode
-    f src
-
-withArg2 f = do
-    (sname : dname : _) <- getArgs
-    src <- openFile sname ReadMode
-    dst <- openFile dname WriteMode
-    f src dst
-```
+Streamly supports reactive and time domain programming inherently because of
+declarative concurrency. See the `Streamly.Prelude` module for some time
+specific combinators like `intervalsOf` and folds like `takeInterval` in
+`Streamly.Internal.Data.Fold`.  Also see pre-release sampling combinators in
+the `Streamly.Internal.Data.Stream.IsStream.Top` module including `throttle`
+and `debounce` like operations.
 
-### cat
+See
+[AcidRain.hs](https://github.com/composewell/streamly-examples/tree/master/examples/AcidRain.hs)
+and
+[CirclingSquare.hs](https://github.com/composewell/streamly-examples/tree/master/examples/CirclingSquare.hs).
 
-``` haskell
-cat = S.fold (FH.writeChunks stdout) . S.unfold FH.readChunks
-main = withArg cat
-```
+## More examples
 
-### cp
+Many more examples can be found in the [streamly-examples
+repository](https://github.com/composewell/streamly-examples).
 
-``` haskell
-cp src dst = S.fold (FH.writeChunks dst) $ S.unfold FH.readChunks src
-main = withArg2 cp
-```
+Streamly comes equipped with a very powerful set of abstractions to accomplish
+any kind of programming tasks that you may want to throw at it. It provides,
+streams, arrays, file-io, fsnotify, network-io, time domain programming
+(reactive programming). See the [streamly
+documentation](https://streamly.composewell.com) to know more.
 
-### wc -l
+## Concurrency
 
-``` haskell
-wcl = S.length . S.splitOn (== 10) FL.drain . S.unfold FH.read
-main = withArg wcl >>= print
-```
+Streamly uses lock-free synchronization for concurrent operations. It employs
+auto-scaling of the degree of concurrency based on demand. For CPU bound tasks
+it tries to keep the threads close to the number of CPUs available whereas for
+IO bound tasks more threads can be utilized. Parallelism can be utilized with
+little overhead even if the task size is very small.  See [concurrency
+benchmarks](https://github.com/composewell/concurrency-benchmarks) for detailed
+performance results and a comparison with the `async` package.
 
-### Average Line Length
+## Performance
 
-``` haskell
-avgll =
-      S.fold avg
-    . S.splitOn (== 10) FL.length
-    . S.unfold FH.read
+As you have seen above in the word count example, streamly enables
+highly modular abstractions with the best possible performance (close to
+an equivalent C program).
 
-    where avg      = (/) <$> toDouble FL.sum <*> toDouble FL.length
-          toDouble = fmap (fromIntegral :: Int -> Double)
+For example, none of the existing Haskell libraries can work with byte
```

Comment:
Although true, don't know how people will take it.

Reply:
This is a key differentiation.